cluster_experiments/power_analysis.py

```diff
@@ -527,6 +547,24 @@ def check_inputs(self):
         self.check_clusters()
 
 
+class PowerAnalysisWithPreExperimentData(PowerAnalysis):
```

Comment:
I'm thinking about what to do to protect the user to use this in the wrong cases. Perhaps we should add a check like
`hasattr(self.analysis, "_split_pre_experiment_df")` (and raise error if false)?

Or derive the synethetic control from PreExperimentAnalysis
```
class SyntheticControlAnalysis(ExperimentAnalysis, PreExperimentAnalysis):
```

and do something like `isinstance(self.analysis, PreExperimentAnalysis)`, and raise error if false.

wdyt?

Reply:
Yes I agree that this can be confusing. What I am thinking now is to rename to something more related to Synthetic control (PowerAnalysisSyntheticControl). Once another estimator uses this class than we can think again of a better naming. 

The best solution would be to wrap both into one class. The user instantiate the class and the behavior will depend on the analysis

Reply:
yep makes sense